topdown/http.go

```diff
@@ -729,13 +772,13 @@ func newHTTPSendCache() *httpSendCache {
 }
 
 func valueHash(v util.T) int {
-	return v.(ast.Value).Hash()
+	return ast.StringTerm(v.(ast.Value).String()).Hash()
```

Comment:
@ashutosh-narkar fyi needed changes to the `hash` and `eq` functions of the hashmap to bring parity between inter and intra query cache key generation. The inter query cache already relies on the `String()` representation of `Value` for cache key.
Relying on the raw `Value` instead of the this representation will always lead to a cache miss in intra query due to change in `Hash` when `Copy()` from request object is performed during key generation.

Reply:
Thanks for the explanation. 

> The inter query cache already relies on the String() representation of Value for cache key.

Can you point to where this code is?

Reply:
@ashutosh-narkar please refer [topdown/cache/cache.go#L240](https://github.com/open-policy-agent/opa/blob/02c565a982e1a1f280712f66704c888f0ad7aa60/topdown/cache/cache.go#L240), `String()` representation of the input key (of type `ast.Value`) is used as a cache key.